specs/deneb/beacon-chain.md

```diff
@@ -130,6 +131,7 @@ class ExecutionPayload(Container):
     withdrawals: List[Withdrawal, MAX_WITHDRAWALS_PER_PAYLOAD]
     data_gas_used: uint64  # [New in Deneb:EIP4844]
     excess_data_gas: uint64  # [New in Deneb:EIP4844]
+    parent_beacon_block_root: Root  # [New in Deneb:EIP4788]
```

Comment:
Before merging, I would like us to clarify why this duplication is reasonable. Alternatively, CL can get `parentRoot` from a beacon block and supply it to EL via Engine API.

I do admit that the complexity is little, but by default it would be great if we can avoid unreasonable data duplication in the core protocol structures.

Reply:
the reasoning is that this container `ExecutionPayload` is easier to reason about if it mirrors _exactly_ the block on the execution layer (with the one exception of the `block_hash` which I think we can all agree is important enough to qualify for an exception) and this field is required to be in the execution layer block to support asynchronous syncing paradigms like we currently have (and want to keep)

Reply:
I would not stand on that but the mirroring argument sounds rather weak to me when we're considering `parentRoot` as this data is very unlikely to disappear from an outer structure

Reply:
I do think that it's not unreasonable that in certain light modes that you have the `ExecutionPayload` and not the rest of the state.

That said, I don't have a terribly strong argument other than that and looser "mirroring"  arguments

Reply:
An EL light client does not see CL `ExecutionPayload` at all, they process only execution block headers. So, the question can be reduced to CL light clients.

A CL light client _has_ to validate the `ExecutionPayload` against the beacon block root, as that beacon block root is what the sync committee is signing. Since it is more compact to send the entire `BeaconBlockHeader` than to send a merkle proof all the way to the root, CL light clients typically have access to the `parent_root`.

When syncing backwards, though, a beacon API could theoretically be defined that provides solely the `ExecutionPayloadHeader`. A CL light client could then use that API to obtain historical `ExecutionPayload` without the corresponding `BeaconBlockHeader`, and validate it using `parent_hash`. Such an API doesn't currently exist, though. 

Further note that a pure CL light client cannot access the correct `transactions_root` / `withdrawals_root`, due to the MPT vs SSZ discussion (EIP-6404). Therefore, a CL practically has to switch to EL specific APIs for any realistic backward sync. And, once it is using EL specific APIs to obtain historic execution block headers, we are back to the EL light client case that isn't affected by CL `ExecutionPayload` shape.

As we don't have mirroring today, due to the `transactions_root` / `withdrawals_root` differences, I think it is a bit premature to include a duplicate `parent_root` in the `ExecutionPayload`, especially as it is quite difficult to later remove information again. If the EL block header is ever transitioned to SSZ, it could make sense to then repurpose the current `block_hash` field to become the `parent_root` to achieve true mirroring (with the real `block_hash` then being `hash_tree_root(execution_payload)` across both CL and EL).

Reply:
went for removal